src/context/mod.rs

```diff
@@ -263,4 +266,54 @@ impl Context {
     ) -> raw::Status {
         raw::notify_keyspace_event(self.ctx, event_type, event, keyname)
     }
+
+    /// Extracts the first regexp capture
+    fn extract_token_value<'a, 'b>(s: &'a str, reg_exp: &'b str) -> Option<&'a str> {
+        match Regex::new(reg_exp) {
+            Ok(re) => match re.captures(s) {
+                Some(captures) => match captures.get(1) {
+                    Some(m) => Some(m.as_str()),
+                    None => None,
+                },
+                None => None,
+            },
+            Err(_) => None,
+        }
+    }
```

Comment:
No sure why you made this a method of `Context`...
In general, the version parsing is spread out over 3 functions: This one, `get_redis_version_internal` and `Version::from`.
The regex parsing also has some duplication.
What I suggest is: Write 1 function that parses the `info` response directly into a `Version`. You can use 3 capture groups to directly extract the major/minor/patch without later needing to split on `.`. This should be fairly simple and clean.